homeassistant/components/shelly/entity.py

```diff
@@ -269,20 +264,11 @@ def async_setup_entry_rest(
     """Set up entities for REST sensors."""
     coordinator = get_entry_data(hass)[config_entry.entry_id].rest
     assert coordinator
-    entities = []
-    for sensor_id in sensors:
-        description = sensors.get(sensor_id)
-
-        if not coordinator.device.settings.get("sleep_mode"):
-            entities.append((sensor_id, description))
-
-    if not entities:
-        return
```

Comment:
This code was copied from `block` part:
https://github.com/home-assistant/core/blob/3f0be3dfba96d52d24d7e83c279be4c88322a767/homeassistant/components/shelly/entity.py#L74-L78

When building `block ` entities we dynamically loop on sensors which may not exists, however when building `REST` entities we loop on our own list and it doesn't make sense to use `get` for fetching items from our own list.

`if not coordinator.device.settings.get("sleep_mode"):` can never fail since we don't call `async_setup_entry_rest` for sleeping devices